RecoLocalCalo/EcalRecProducers/plugins/alpaka/TimeComputationKernels.h

```diff
@@ -0,0 +1,1162 @@
+
+        if (sample == 0) {
+          // note, subtract to remove the double counting of sample == 3
+          auto const sum0 = s_sum0[ltx] + s_sum0[ltx + 1] - s_sum0[ltx + 3];
+          auto const sum1 = s_sum1[ltx] + s_sum1[ltx + 1] - s_sum1[ltx + 3];
+          auto const sumA = s_sumA[ltx] + s_sumA[ltx + 1] - s_sumA[ltx + 3];
+          auto const sumAA = s_sumAA[ltx] + s_sumAA[ltx + 1] - s_sumAA[ltx + 3];
+          auto const chi2 = sum0 > 0 ? (sumAA - sumA * sumA / sum1) / sum0 : static_cast<ScalarType>(0);
+          chi2s[ch] = chi2;
+          sum0s[ch] = sum0;
+          sumAAs[ch] = sumAA;
+
+#ifdef DEBUG_TC_NULLHYPOT
+          if (ch == 0) {
+            printf("chi2 = %f sum0 = %d sumAA = %f\n", chi2, static_cast<int>(sum0), sumAA);
+          }
+#endif
+        }
+      }
+    }
+  };
+
+  //
+  // launch ctx parameters are
+  // 45 threads per channel, X channels per block, Y blocks
+  // 45 comes from: 10 samples for i <- 0 to 9 and for j <- i+1 to 9
+  // TODO: it might be much beter to use 32 threads per channel instead of 45
+  // to simplify the synchronization
+  class Kernel_time_compute_makeratio {
+    using ScalarType = ::ecal::multifit::SampleVector::Scalar;
+
+  public:
+    template <typename TAcc, typename = std::enable_if_t<alpaka::isAccelerator<TAcc>>>
+    ALPAKA_FN_ACC void operator()(TAcc const& acc,
+                                  EcalDigiDeviceCollection::ConstView digisDevEB,
+                                  EcalDigiDeviceCollection::ConstView digisDevEE,
+                                  ScalarType* const sample_values,
+                                  ScalarType* const sample_value_errors,
+                                  bool* const useless_sample_values,
+                                  char* const pedestal_nums,
+                                  ScalarType* const sumAAsNullHypot,
+                                  ScalarType* const sum0sNullHypot,
+                                  ScalarType* tMaxAlphaBetas,
+                                  ScalarType* tMaxErrorAlphaBetas,
+                                  ScalarType* g_accTimeMax,
+                                  ScalarType* g_accTimeWgt,
+                                  TimeComputationState* g_state,
+                                  EcalMultifitParametersDevice::ConstView paramsDev,
+                                  ConfigurationParameters::type const timeFitLimits_firstEB,
+                                  ConfigurationParameters::type const timeFitLimits_firstEE,
+                                  ConfigurationParameters::type const timeFitLimits_secondEB,
+                                  ConfigurationParameters::type const timeFitLimits_secondEE) const {
+      // constants
+      constexpr uint32_t nchannels_per_block = 10;
+      constexpr auto nthreads_per_channel = nchannels_per_block * (nchannels_per_block - 1) / 2;
+      constexpr auto nsamples = EcalDataFrame::MAXSAMPLES;
```

Comment:
These constants could be moved to be `static constexpr` (or `enum`) at `class`-level.
Then, they could be reused also from the host code to properly size the blocks and threads, without having to duplicate the same definitions there.

Reply:
Can be improved later, see https://github.com/cms-sw/cmssw/issues/43551 .